src/applications/disability-benefits/all-claims/utils.jsx

```diff
@@ -267,6 +268,51 @@ export function transformProviderFacilities(providerFacilities) {
   }));
 }
 
+/**
+ * Concatenates incident location address object into location string. This will ignore null
+ *  or undefined address fields
+ * @param {Object} incidentLocation location address with city, state, country, and additional details
+ * @returns {String} incident location string
+ */
+export function concatIncidentLocationString(incidentLocation) {
+  return [
+    incidentLocation.city,
+    incidentLocation.state,
+    incidentLocation.country,
+    incidentLocation.additionalDetails,
+  ]
+    .filter(locationField => locationField)
+    .join(', ');
+}
+
+/**
+ * Returns an array of the maximum set of PTSD incident form data field names
+ */
+export function getFlatIncidentKeys() {
+  const incidentKeys = [];
+
+  for (let i = 0; i < PTSD_INCIDENT_ITERATION; i++) {
+    incidentKeys.push(`incident${i}`);
+  }
+  for (let i = 0; i < PTSD_INCIDENT_ITERATION; i++) {
+    incidentKeys.push(`secondaryIncident${i}`);
+  }
+
+  return incidentKeys;
+}
+
+export function getPtsdChangeFieldTitles(changeFields, formConfig) {
+  return Object.keys(changeFields)
+    .filter(
+      key =>
+        key !== 'other' &&
+        key !== 'otherExplanation' &&
+        formConfig[key] &&
+        typeof formConfig[key]['ui:title'] === 'string',
```

Comment:
:thinking: This actually has me pretty concerned.

What happens when we forget this little nugget (that all of these check boxes need string titles) and we have to embolden a word later? We'd change the `ui:title` to a react component and add a `<strong>` tag. This will work without failing anywhere down the line, but when we go to send the data to the api, if a user selects that option with the emboldened word, we'll filter it out here.

This means the option they selected won't make it to the PDF and we'll never be notified.

Grabbing the text from the `ui:title` to send to the PDF makes me a bit uneasy, but it's _mostly_ okay. We just need to protect our future selves for when we forget these all need to be strings.

Can you make a unit test that ensures all the fields that run through this function have a string `ui:title`? If we have to change this up later to embolden a word or something, we can refactor this bit, but I want to make sure we catch it in the first place.

Reply:
I moved the titles to `constants` so it can be referenced by the transform and the ui, but the ui would be able to override the value in the UI if it ever came up. I also added a test to validate that the mapped change labels are strings.